extension/background/languages.js

```diff
@@ -0,0 +1,32 @@
+this.languages = {};
+
+this.languageList = (function() {
+    const exports = {};
+
+    const LANGUAGES = {
+        Czech: "cs",
+        Danish: "da",
+        Dutch: "nl",
+        English: "en",
+        Finnish: "fi",
+        French: "fr",
+        German: "de",
+        Hungarian: "hu",
+        Italian: "it",
+        Norwegian: "no",
+        Polish: "pl",
+        Portuguese: "pt",
+        Romanian: "ro",
+        Russian: "ru",
+        Slovak: "sk",
+        Slovenian: "sl",
+        Spanish: "es",
+        Swedish: "sv",
+        Turkish: "tr",
+        Ukrainian: "uk",
+    };
+
+    exports.allLanguages = function() {
+        return Object.keys(LANGUAGES);
+    };
+})();
```

Comment:
This needs `return exports` at the end, that's your basic problem.

Reply:
I fixed the most of it, but I am still not sure how should my languages.js file look like, since this way it's not working.

```
this.languages = (function() {
    const exports = {};

    exports.languageCode = {
        Czech: "cs",
        Danish: "da",
        Dutch: "nl",
        English: "en",
        Finnish: "fi",
        French: "fr",
        German: "de",
        Hungarian: "hu",
        Italian: "it",
        Norwegian: "no",
        Polish: "pl",
        Portuguese: "pt",
        Romanian: "ro",
        Russian: "ru",
        Slovak: "sk",
        Slovenian: "sl",
        Spanish: "es",
        Swedish: "sv",
        Turkish: "tr",
        Ukrainian: "uk",
    };

    exports.getLanguages = function() {
        return exports;
    };
})();
```

Or maybe I am calling it the wrong way in the **intentParser.js** :
```
const ENTITY_TYPES = {
    serviceName: serviceList.allServiceNames(),
    musicServiceName: serviceList.musicServiceNames(),
    lang: languages.getLanguages(),
  };
```

Reply:
Remember the whole module is a function, and `this.languages` is the result of that function – so you need `return exports;` right at the end of the function. As it is the function isn't returning anything, and so `this.languages` is undefined.

Reply:
Alright, I think it's finally working.
![Screenshot_1](https://user-images.githubusercontent.com/44289095/68944465-2aae6e80-07ae-11ea-8246-daf8c63c01e4.png)

One more question. Shall I give this functionality to the `navigation.translateSelection` intent too?

Reply:
Yes, all this makes sense for translateSelection as well

Reply:
Okay, done. Are we finished?